crates/xlayer/db/src/internal_transaction_inspector.rs

```diff
@@ -0,0 +1,307 @@
+        txn.is_error = false;
+        txn.gas = gas_limit;
+        txn.value_wei = if value_wei.is_empty() { "0" } else { &value_wei }.to_string();
+        txn.call_value_wei = if let Ok(value) = value_wei.parse::<u128>() {
+            format!("0x{:x}", value)
+        } else {
+            String::from("0x0")
+        };
+
+        txn.to = to.clone();
+        match txn.call_type.as_str() {
+            "delegatecall" => {
+                txn.from = to;
+                txn.to = code_address;
+                txn.trace_address = txn.from.clone();
+            }
+            "callcode" => {
+                txn.code_address = code_address;
+            }
+            _ => {}
+        }
+    }
+
+    fn before_op(&mut self) {
+        let depth = self.current_path.len();
+
+        match depth.cmp(&self.last_depth) {
+            Ordering::Greater => {
+                self.sibling_count.push(0);
+            }
+            Ordering::Less => {
+                self.sibling_count.truncate(depth + 1);
+            }
+            Ordering::Equal => {}
+        }
+        self.last_depth = depth;
+
+        let internal_index = self.sibling_count[depth];
+        let trace_index = self.traces.len() - 1;
+        self.trace_stack.push(trace_index);
+
+        let txn = &mut self.traces[trace_index];
+        txn.dept = depth as u64;
+        txn.internal_index = internal_index as u64;
+
+        self.sibling_count[depth] += 1;
+
+        self.current_path.push(internal_index);
+        if self.current_path.len() > 1 {
+            let suffix =
+                self.current_path[1..].iter().map(|s| s.to_string()).collect::<Vec<_>>().join("_");
+
+            txn.name.reserve(1 + suffix.len());
+            txn.name.push('_');
+            txn.name.push_str(&suffix);
+        }
+
+        txn.name = txn.call_type.clone() + txn.name.as_str();
+    }
+
+    fn after_op(&mut self) {
+        self.current_path.pop();
+        if self.trace_stack.is_empty() {
+            self.all_traces.push(self.traces.clone());
+            self.reset();
+        }
+    }
+
+    pub fn get(&mut self) -> Vec<Vec<InternalTransaction>> {
+        self.all_traces.clone()
+    }
+
+    pub fn reset(&mut self) {
+        self.traces.clear();
+        self.current_path.clear();
+        self.last_depth = 0;
+        self.sibling_count = vec![0];
+        self.trace_stack.clear();
+    }
+}
+
+impl<CTX> Inspector<CTX, EthInterpreter> for TraceCollector
+where
+    CTX: ContextTr,
+{
+    fn call(&mut self, ctx: &mut CTX, inputs: &mut CallInputs) -> Option<CallOutcome> {
+        let call_type = match inputs.scheme {
+            reth_revm::interpreter::CallScheme::Call => "call".to_string(),
+            reth_revm::interpreter::CallScheme::CallCode => "callcode".to_string(),
+            reth_revm::interpreter::CallScheme::DelegateCall => "delegatecall".to_string(),
+            reth_revm::interpreter::CallScheme::StaticCall => "staticcall".to_string(),
+        };
```

Comment:
```suggestion
        let call_type = match inputs.scheme {
            reth_revm::interpreter::CallScheme::Call => "call",
            reth_revm::interpreter::CallScheme::CallCode => "callcode",
            reth_revm::interpreter::CallScheme::DelegateCall => "delegatecall",
            reth_revm::interpreter::CallScheme::StaticCall => "staticcall",
        }
        .to_string();
```
Can use `to_string` once